File: client/src/api/runs.js
/** Run history API - contract: server/app/schemas/run.py (BE: Person 3, FE: Person 2). */
import { isDemoMode } from '@/config';
import runsMock from '@/mocks/runs_mock.json';
import { http } from './http';
import { demo } from './demo';

export const runsApi = {
  list(projectId) {
    if (isDemoMode) return demo({ ...runsMock, project_id: projectId });
    return http.get(`/api/v1/projects/${projectId}/runs`);
  },

  get(projectId, runId) {
    if (isDemoMode) return demo(runsMock.runs.find((r) => r.run_id === runId) ?? runsMock.runs[0]);
    return http.get(`/api/v1/projects/${projectId}/runs/${runId}`);
  },
};

File: server/app/api/runs.py
"""Run history endpoints. Owner: Person 3 | BL-QA-03."""

from fastapi import APIRouter

from app.schemas.run import RunListResponse, RunRecord
from app.services import run_store

router = APIRouter(prefix="/api/v1/projects/{project_id}/runs", tags=["runs"])


@router.get("", response_model=RunListResponse)
async def list_runs(project_id: str) -> RunListResponse:
    return RunListResponse(project_id=project_id, runs=run_store.list_runs(project_id))


@router.get("/{run_id}", response_model=RunRecord)
async def get_run(project_id: str, run_id: str) -> RunRecord:
    return run_store.get_run(project_id, run_id)
